### DISCUSSION TITLE
How to clear `run_once_` cache? Or force `run_once_` to run many times without edit.

### DISCUSSION THREAD

**Discussion by narze** [twpayne/chezmoi#1678] [Q&A] [answered]
Now I'm moving from dotbot to chezmoi and I love it 🤩 
However as I'm new to templating the dotfiles, I have some of the scripts that should be run only once per change per machine e.g. `brew install` or installing Fish shell
In the development step, how can I clear `run_once_` cache to run it many times without editing? (Now I have to work-around by adding whitespaces to the file, then run `chezmoi apply [filename]` again.)

**Comment by twpayne** [comment, marked as answer]
You can use `chezmoi state reset` to reset the **whole** state, see the [`state` command](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#state) for a bit more info. chezmoi stores the state of `run_once_` scripts in the `scriptState` bucket, with the SHA256 of the script's contents as the key. Right now, there's not an easy way to clear the `run_once_` state for a single script.

**Reply by twpayne** [reply]
Once #1680 is merged and released you will also be able to run:
```console
$ chezmoi delete-bucket --bucket=scriptState
```

**Reply by narze** [reply]
Thank you!

**Reply by miles170** [reply]
The command has now been replaced with:

```
$ chezmoi state delete-bucket --bucket=scriptState
```

**Comment by nandalopes** [comment]
When I need to re-run a `run_once` script, I usally use `chezmoi cat <script filename without run_once>` and pipe to script exec

**Reply by nandalopes** [reply]
For example, to re-run [run_onchange_after_post_install_yadr](https://github.com/nandalopes/dotfiles/blob/main/root/.chezmoiscripts/scripts/run_onchange_after_post_install_yadr.sh.tmpl):

```console
chezmoi cat ~/.chezmoiscripts/post_install_yadr.sh | bash
```

**Reply by alexdelprete** [reply]
Hi @nandalopes,

I was looking for a way to manually run my `run_onchange_install-packages.sh.tmpl` that lives in the source state folder without resetting the state, and luckily found your post.

Thank you so much, and thanks also to @twpayne for creating chezmoi. :)
